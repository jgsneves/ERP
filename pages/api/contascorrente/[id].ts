import { NextApiRequest, NextApiResponse } from "next";
import prisma from "../../../services/Prisma";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  switch (req.method) {
    case "GET":
      try {
        const { id } = req.query;

        const result = await prisma.contasCorrente.findUnique({
          where: {
            Id: id as string,
          },
        });

        if (result) {
          res.json(result);
          return;
        }

        res.status(404).send({ message: "Conta corrente não encontrada." });
      } catch (error) {
        res.status(500).send({ error });
      }
      break;

    case "DELETE":
      try {
        const { id } = req.query;

        const result = await prisma.contasCorrente.delete({
          where: {
            Id: id as string,
          },
        });

        res.json(result);
      } catch (error) {
        res.status(500).send({ error });
      }

      break;
    default:
      res.status(400).send({
        metodo: req.method,
        message: "Método não implementado para esta entidade",
      });
      break;
  }
}
