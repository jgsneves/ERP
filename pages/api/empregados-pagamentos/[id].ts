import { NextApiRequest, NextApiResponse } from "next";
import prisma from "services/Prisma";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  switch (req.method) {
    case "PATCH":
      try {
        const result = await prisma.empregadosPagamentos.update({
          data: req.body,
          where: {
            Id: req.query.id as string,
          },
        });
        res.json(result);
      } catch (error) {
        console.log({ error });
        res.status(500).send({ error });
      }
      break;

    default:
      res.status(400).send({
        method: req.method,
        message: "Método não implementado para esta entidade!",
      });
      break;
  }
}
