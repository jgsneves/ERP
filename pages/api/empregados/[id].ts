import { NextApiRequest, NextApiResponse } from "next";
import prisma from "../../../services/Prisma";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  switch (req.method) {
    case "GET":
      try {
        const result = await prisma.pessoas.findUnique({
          where: {
            Id: req.query.id as string,
          },
          include: {
            ContasCorrentes: true,
            Endereco: true,
          },
        });

        if (result) return res.json(result);

        res.status(404).send({ message: "Empregado não encontrado!" });
      } catch (error) {
        console.log({ error });

        res.status(500).send({ error });
      }
      break;

    case "PATCH":
      try {
        await prisma.pessoas.update({
          where: {
            Id: req.query.id as string,
          },
          data: req.body,
        });
        res.status(200).send({ message: "Empregado atualizado com sucesso! " });
      } catch (error) {
        res
          .status(500)
          .send({ message: "Não foi possível editar este empregado!" });
      }
      break;

    default:
      res.status(400).send({
        method: req.method,
        message: "Método não implementado para esta entidade!",
      });
      break;
  }
}
