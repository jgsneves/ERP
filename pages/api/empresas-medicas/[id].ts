import { NextApiRequest, NextApiResponse } from "next";
import prisma from "services/Prisma";
import { ErrorHandler } from "utils/ErrorHandler";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  switch (req.method) {
    case "GET":
      try {
        const result = await prisma.empresasMedicas.findUnique({
          where: {
            Id: req.query.id as string,
          },
          include: {
            Socios: true,
          },
        });

        if (result) return res.json(result);

        res.status(404).send({ message: "Empresa médica não encontrada! " });
      } catch (error) {
        ErrorHandler.logPrismaError(error);
        res.status(500).send({ error });
      }
      break;

    default:
      res.status(400).send({
        metodo: req.method,
        message: "Método não implementado para esta entidade!",
      });
      break;
  }
}
