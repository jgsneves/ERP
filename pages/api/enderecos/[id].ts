import { NextApiRequest, NextApiResponse } from "next";
import prisma from "services/Prisma";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  switch (req.method) {
    case "GET":
      try {
        const { id } = req.query;
        const result = await prisma.enderecos.findUnique({
          where: {
            Id: id as string,
          },
        });
        if (result) {
          res.json(result);
          return;
        }
        res.status(404).send({ message: "Endereço não encontrado." });
      } catch (error) {
        res.status(500).send({ error });
      }
      break;

    case "PUT":
      try {
        const { body, query } = req;
        const result = await prisma.enderecos.update({
          where: {
            Id: query.id as string,
          },
          data: body,
        });
        res.status(200).send(result);
      } catch (error) {
        res.status(500).send({ error });
      }
      break;

    default:
      res.status(500).send({
        metodo: req.method,
        message: "Método não implementado para esta entidade!",
      });
      break;
  }
}
