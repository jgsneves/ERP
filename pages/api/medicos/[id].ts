import { NextApiRequest, NextApiResponse } from "next";
import prisma from "services/Prisma";
import { ErrorHandler } from "utils/ErrorHandler";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  switch (req.method) {
    case "GET":
      try {
        const result = await prisma.pessoas.findUnique({
          where: {
            Id: req.query.id as string,
          },
        });

        if (result) {
          return res.json(result);
        }

        res.status(404).send({ message: "Médico não encontrado." });
      } catch (error) {
        ErrorHandler.logPrismaError(error);
        res.status(500).send({ error });
      }
      break;

    case "PATCH":
      try {
        const requestBody = req.body;

        await prisma.pessoas.update({
          where: {
            Id: req.query.id as string,
          },
          data: requestBody,
        });

        res.status(200).send({ message: "Médico atualizado com sucesso!" });
      } catch (error) {
        ErrorHandler.logPrismaError(error);
        res.status(500).send({ error });
      }
      break;
    default:
      res.status(400).send({
        method: req.method,
        message: "Método não implementado para esta entidade!",
      });
      break;
  }
}
