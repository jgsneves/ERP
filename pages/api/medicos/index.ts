import { Pessoas, PessoasTipo } from "@prisma/client";
import { NextApiRequest, NextApiResponse } from "next";
import prisma from "services/Prisma";
import { ErrorHandler } from "utils/ErrorHandler";

interface EmpresaMedicaRazaoSocial {
  RazaoSocial: string;
}

interface Doctors extends Pessoas {
  EmpresaMedica: EmpresaMedicaRazaoSocial | null;
}

export interface MedicosResponse {
  pagina: number;
  totalPaginas: number;
  doctors: Doctors[];
}

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  switch (req.method) {
    case "GET":
      try {
        const pagina = parseInt(req.query.pagina as string) || 1;
        const quantidade = parseInt(req.query.quantidade as string) || 10;
        const empresaMedicaId = req.query.empresaMedicaId;

        const whereClause = {
          EmpresaMedicaId:
            typeof empresaMedicaId === "string" ? empresaMedicaId : undefined,
          Tipo: PessoasTipo.MEDICO,
        };

        const doctors = await prisma.pessoas.findMany({
          skip: (pagina - 1) * quantidade,
          take: quantidade,
          orderBy: {
            Nome: "asc",
          },
          where: whereClause,
          include: {
            EmpresaMedica: {
              select: {
                RazaoSocial: true,
              },
            },
          },
        });

        const count = doctors.length;
        const totalPaginas = Math.ceil(count / quantidade);
        res.json({ pagina, totalPaginas, doctors });
      } catch (error) {
        ErrorHandler.logPrismaError(error);
        res.status(500).send({ error });
      }
      break;

    case "POST":
      try {
        const result = await prisma.pessoas.create({
          data: req.body,
        });
        res.json(result);
      } catch (error) {
        ErrorHandler.logPrismaError(error);
        res.status(500).send({ error });
      }
      break;
    default:
      res.status(400).send({
        metodo: req.method,
        message: "Método não implementado para esta entidade!",
      });
      break;
  }
}
