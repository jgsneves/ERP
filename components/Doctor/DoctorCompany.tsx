import { Badge, Spinner, Text, VStack } from "@chakra-ui/react";
import { EmpresasMedicas } from "@prisma/client";
import ContentTitle from "components/Shared/ContentTitle";
import { DoctorEntity } from "pages/medicos/[id]";
import useSwr from "swr";
import { fetcher } from "utils/fetcher";
import DoctorSearchEmpresaMedica from "./DoctorSearchEmpresaMedica";

interface Props {
  doctor: DoctorEntity;
}

export default function DoctorCompany({ doctor }: Props) {
  const { data, isLoading, error } = useSwr<EmpresasMedicas>(
    doctor.EmpresaMedicaId
      ? `/api/empresas-medicas/${doctor.EmpresaMedicaId}`
      : null,
    fetcher,
    {
      revalidateOnFocus: false,
    }
  );

  if (isLoading) return <Spinner />;

  if (error) return <Text>Houve um erro na busca da empresa médica.</Text>;

  return (
    <VStack w="300px" alignItems="flex-start">
      <ContentTitle title="Vinculado à empresa" />
      {doctor.EmpresaMedicaId ? (
        <Badge colorScheme="blue">
          {doctor.EmpresaMedicaId ? data?.RazaoSocial : "N/A"}
        </Badge>
      ) : (
        <DoctorSearchEmpresaMedica doctor={doctor} />
      )}
    </VStack>
  );
}
