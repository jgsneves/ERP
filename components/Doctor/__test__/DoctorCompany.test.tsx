import { EmpresasMedicas } from "@prisma/client";
import { render } from "@testing-library/react";
import { DoctorEntity } from "pages/medicos/[id]";
import DoctorCompany from "../DoctorCompany";

jest.mock("swr", () => {
  const originalModule = jest.requireActual("swr");

  const firstResponse: EmpresasMedicas = {
    Cnpj: "123",
    CriadoEm: new Date(),
    EnderecoId: null,
    Id: "1",
    ModificadoEm: null,
    RazaoSocial: "ABCD Empresa Médica",
  };

  return {
    __esModule: true,
    ...originalModule,
    default: jest
      .fn()
      .mockReturnValueOnce({
        isLoading: false,
        data: firstResponse,
      })
      .mockReturnValueOnce({
        isLoading: false,
        data: undefined,
        error: { ble: "foo" },
      })
      .mockReturnValue({ isLoading: false, data: undefined }),
  };
});

jest.mock("next/router", () => {
  return {
    useRouter: jest.fn(),
  };
});

describe("DoctorCompany test suite", () => {
  it("should show badge", () => {
    const mockedProp = {
      EmpresaMedicaId: "1",
    } as DoctorEntity;
    const { getByText, asFragment } = render(
      <DoctorCompany doctor={mockedProp} />
    );
    expect(getByText("ABCD Empresa Médica")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should show error", () => {
    const mockedProp = {} as DoctorEntity;
    const { queryByText, asFragment } = render(
      <DoctorCompany doctor={mockedProp} />
    );
    expect(queryByText("ABCD Empresa Médica")).toBeNull();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should not render a badge", () => {
    const mockedProp = {} as DoctorEntity;
    const { queryByText, asFragment } = render(
      <DoctorCompany doctor={mockedProp} />
    );
    expect(queryByText("ABCD Empresa Médica")).toBeNull();
    expect(asFragment()).toMatchSnapshot();
  });
});
