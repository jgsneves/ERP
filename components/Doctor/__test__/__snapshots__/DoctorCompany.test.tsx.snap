// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoctorCompany test suite should not render a badge 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-1av7kul"
  >
    <div
      class="css-8v4id7"
    >
      <b
        class="chakra-text css-r262sy"
      >
        Vinculado à empresa
      </b>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-0"
      >
        Vincule este médico a uma empresa médica:
      </p>
      <input
        class="chakra-input css-0"
        placeholder="procure uma empresa por nome"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DoctorCompany test suite should show badge 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-1av7kul"
  >
    <div
      class="css-8v4id7"
    >
      <b
        class="chakra-text css-r262sy"
      >
        Vinculado à empresa
      </b>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
    </div>
    <span
      class="chakra-badge css-ifsvay"
    >
      ABCD Empresa Médica
    </span>
  </div>
</DocumentFragment>
`;

exports[`DoctorCompany test suite should show error 1`] = `
<DocumentFragment>
  <p
    class="chakra-text css-0"
  >
    Houve um erro na busca da empresa médica.
  </p>
</DocumentFragment>
`;
